resources/aws_ec2.go

```diff
@@ -677,3 +681,40 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServe returns an http server used to listen for sns messages.
+func (obj *AwsEc2Res) snsServe() *http.Server {
+	addr := obj.WatchListenAddr
+	// if addr is a port
+	if _, err := strconv.Atoi(obj.WatchListenAddr); err == nil {
+		addr = fmt.Sprintf(":%s", obj.WatchListenAddr)
+	}
+	srv := &http.Server{Addr: addr}
+	log.Printf("%s: Starting SNS Endpoint", obj)
+	http.HandleFunc("/", obj.postHandler)
+	log.Printf("%s: Listening on %s", obj, addr)
+	obj.wg.Add(1)
+	go func() {
```

Comment:
so this part needs some discussion...
i'm tempted to say that this `run` part of the server should be a different method to keep the awsChan stuff out of this.
eg: this should *only* build the server... meaning we might rename to `snsServer` for example.
perhaps we could merge this without this goroutine until it's clear how best it should get called?
happy to review that as a second patch in this PR if that's what comes next. LMK what you think. (don't just blindly agree with me, disagreeing is fine)

Reply:
I think that's fine. Separation of concerns and all that. I'll remove it from this commit, and add a second one with the listener.